llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -6945,7 +6945,16 @@ const ConstantRange &ScalarEvolution::getRangeRef(
         ConstantRange RangeFromOps(BitWidth, /*isFullSet=*/false);
 
         for (const auto &Op : Phi->operands()) {
-          auto OpRange = getRangeRef(getSCEV(Op), SignHint, Depth + 1);
+          ConstantRange OpRange = RangeFromOps;
+          if (auto *S = getExistingSCEV(Op)) {
+            OpRange = getRangeRef(S, SignHint, Depth + 1);
+          } else {
+            if (!Op->getType()->isIntOrIntVectorTy())
+              break;
+            SimplifyQuery SQ(DL, &DT, &AC, Phi, true);
```

Comment:
This should use the terminator of the incoming block instead of the phi as context instruction.

Reply:
This patch currently fails to bootstrap clang. Might be because of the wrong context... or some other problem.

Reply:
Looks like some polly lit tests are failing?